src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -210,44 +219,46 @@ public LocalRepository localRepositoryFor(GitHubProject project) throws IOExcept
   private LocalRepository loadLocalRepositoryFor(GitHubProject project) throws IOException {
     Objects.requireNonNull(project, "On no! Project is null!");
 
-    LocalRepositoryInfo info = localRepositoriesInfo.get(project.url());
-    if (info == null) {
-      Path repositoryPath = base
-          .resolve(project.organization().name())
-          .resolve(project.name());
+    synchronized (localRepositoriesInfo) {
```

Comment:
I am afraid that using a monitor of the `localRepositoriesInfo` field is not enough. The `loadLocalRepositoryFor()` method accesses cloned repositories and keep info about them in the `localRepositoriesInfo` map. When an instance of `GitHubDataFetcher` is used in multiple threads, then a lock on `localRepositoriesInfo` is going to protect the map from a race. However, when we have multiple instances of `GitHubDataFetcher`, then calling the`loadLocalRepositoryFor()` method on them is going to result concurrent access to the filesystem. I think the method should use a monitor of a static constant that is shared between all instances of the class. Or, the method can use a lock-file in the filesystem.

Reply:
Perhaps a bit late to ask this. 
But do you think about making it thread-safe for use as singleton only would be better?
In combination with adding according static method to get (create) the singleton instance?

Reply:
I am not sure what you mean and why it is late. Do you suggest to make it a singleton?

I suggest to use the following:

```
private static final Object localRepositoryLock = new Object();
...
synchronized (localRepositoryLock) {
  ... work with local repositories
}
```

Reply:
No, I do not see a problem. 
Just thought make `GitHubDataFetcher` is less effort 😄 
Will update it accordingly.

Reply:
Sure.

> Just thought make GitHubDataFetcher is less effort

Sorry, didn't get it. To make it what?

Reply:
Sorry, missed _singleton_

Reply:
No, I am not sure if making it a singleton would be less effort. This change would result to updating all data provider and their tests.

Reply:
Yes, you are right. Have not thought about this.